unit/solvers/refinement/string_constraint_instantiation/instantiate_not_contains_.cpp

```diff
@@ -0,0 +1,506 @@
+/*******************************************************************\
+
+ Module: Unit tests for `instantiate_not_contains_`.
+
+ Author: Jesse Sigal, jesse.sigal@diffblue.com
+
+\*******************************************************************/
+
+#include <catch.hpp>
+
+#include <solvers/refinement/string_constraint_instantiation.h>
+
+#include <solvers/sat/satcheck.h>
+#include <solvers/refinement/bv_refinement.h>
+#include <java_bytecode/java_types.h>
+#include <langapi/mode.h>
+#include <java_bytecode/java_bytecode_language.h>
+#include <util/namespace.h>
+#include <util/symbol_table.h>
+#include <util/simplify_expr.h>
+
+/// \class Types used throughout the test. Currently it is impossible to
+/// statically initialize this value, there is a PR to allow this
+/// diffblue/cbmc/pull/1213
+class tt
+{
+public:
+  tt() {}
+  typet char_type() const {return java_char_type();}
+  typet length_type() const {return java_int_type();}
+  array_typet array_type() const
+  {
+    return array_typet(char_type(), infinity_exprt(length_type()));
+  }
+  refined_string_typet string_type() const
+  {
+    return refined_string_typet(length_type(), char_type());
+  }
```

Comment:
You can't currently have statically initialized objects that contain `dstrings`, and so I can't have constants at the top of the file, which is what I really want. However, there's a PR which may get merged diffblue/cbmc/pull/1213 which will fix this. This file will be expanded at some point, so having the string and witness types be constant but still parameterized by the length and char types seems desirable, and this way gives the least overhead I feel.